### ISSUE TITLE
Mesh_3: segfault in the Polyhedron demo

### ISSUE THREAD

**Issue by afabri** [CGAL/cgal#2523]
## Issue Details

In the Polyhedron demo:

- Load `bull.off`
- Remove the self intersection
- 3D Tetrahedral Mesh Generation -> Surface mesh generation
- Do not protect sharp edges
- Divide the approximation error by 10
- Advanced: tick manifold

mesh -> crash

@lrineau please have a look

**Comment by lrineau** [comment]
The manifold criterion can work on input with sharp features, if the features are not protected. Here is the camera position showing the issue in the bull:
```
0.268352 -0.267889 -0.189292 0.177328 0.593829 -0.00561001 0.784787
```

**Comment by sloriot** [comment]
@lrineau I'm a bit surprised that a crash of the demo is not a bug. Moreover I don't understand your answer provided the bug report says no sharp edges were protected.

**Comment by lrineau** [comment]
A Delaunay refinement algorithm cannot mesh a wedge without protecting the sharp edge of the wedge. If you impose the manifold criterion, then the mesh algorithm will keep inserting new vertices around that edge, until the crash.

As for the demo crashing, if we compile with CGAL assertions, then the demo will show a popup corresponding to the assertion. But without assertion, I do not see any way to ensure that the user fulfill the preconditions.

**Comment by MaelRL** [comment]
Might be confused because https://github.com/CGAL/cgal/issues/2523#issuecomment-335774752 has a typo and says `The manifold criterion can work [...]`.

**Comment by afabri** [comment]
It might make sense to finalize some code that stops the refinement process when triangles start getting too small, and then ignore requested quality of elements locally.
